crates/core/src/host/wasmtime/pooling_stack_creator.rs

```diff
@@ -0,0 +1,81 @@
+use core::mem::{ManuallyDrop, MaybeUninit};
+use crossbeam_queue::ArrayQueue;
+use std::sync::{Arc, Weak};
+use wasmtime::{StackCreator, StackMemory};
+use wasmtime_internal_fiber::FiberStack;
+
+pub struct PoolingStackCreator {
+    /// The actual pool of stacks.
+    pool: ArrayQueue<FiberStack>,
+    /// A weak reference to `self` that can be cloned
+    /// and put into a `PooledFiberStack` which uses `weak` on drop.
+    /// We do it self-referentially so that we can avoid another indirection.
+    weak: MaybeUninit<Weak<PoolingStackCreator>>,
```

Comment:
I don't think this needs to be `MaybeUninit`? You can just use `new_cyclic`.

Reply:
Yep, I forgot to clean this part up. 😊